tests/vector/test_sync_vector_env.py

```diff
@@ -167,3 +167,17 @@ def test_custom_space_sync_vector_env():
         "step(action-5)",
         "step(action-7)",
     )
+
+
+def test_sync_vector_env_seed():
+    env = make_env("BipedalWalker-v3", seed=123)()
```

Comment:
I suggest parametrizing the env name to get some different types of spaces in.
There's also some space fixture you can reuse from the vectorenv tests, so you also cover batching of custom spaces as well.

Reply:
Part of the problem, related to the testing batch space seeds is that the batched space of Discrete and MultiDiscrete do not produce the same randomness as the original space. For Discrete, MultiDiscrete can be modified to produce the same randomness but that didn't feel part of the original PR. 
This means that we can only test this PR with certain environments (i.e. ones without Discrete or MultiDiscrete action spaces). 
The previous test (test_batched_space_seed) should test that seeding works for all of the spaces and that sample works. So I dont see much point testing more here. Am I missing something?

Reply:
> Part of the problem, related to the testing batch space seeds is that the batched space of Discrete and MultiDiscrete do not produce the same randomness as the original space

Yes, but that's perfectly ok. What you need to test is that the randomness of the batched space is deterministic, given the initial seed.

Something like:

```python
@pytest.mark.parametrize("space", <lots_of_diverse_spaces>)
@pytest.mark.parametrize("n", [1, 2, 5])
@pytest.mark.parametrize("base_seed", [123, 456])
def test_deterministic(space: Space, n: int, base_seed: int):
    space_a = space
    space_a.seed(base_seed)
    space_b =copy.deepcopy(space_a)
    
    # Use the rng of space_a, just to be able to check below that space_a_batched
    # doesn't depend on the original rng of space_a:
    space_a.sample()
    
    space_a_batched = batch_space(space_a, n)
    space_b_batched = batch_space(space_b, n)
    
    assert space_a_batched == space_b_batched
    # Note: Use something from https://numpy.org/doc/stable/reference/routines.testing.html
    # to compare deeply nested objects/dicts.
    assert space_a_batched.sample() == space_b_sample.sample()
```

>This means that we can only test this PR with certain environments (i.e. ones without Discrete or MultiDiscrete action spaces).

I think that's Incorrect. See above, plus, nested / deeply nested dicts, tuples, and custom spaces also need to be tested.

> The previous test (test_batched_space_seed) should test that seeding works for all of the spaces and that sample works. So I dont see much point testing more here. Am I missing something?

Yes:
- I believe there should also be a test to assert that the RNG from the original (unbatched) space isn't shared with the batched space. (see the comment in the snippet above)

Reply:
Also, forgot to include it in my snippet above, but it would also be important to check explicitly that different seeds give different results in the batched spaces too.

Reply:
Hey @pseudo-rnd-thoughts, would you mind un-resolving the conversation that was here? I don't seem to be able to do so myself.

Reply:
Still need to parametrize the env name here IMO.

Reply:
Done in b7fc60fdb8a0dc77e8615ffc18dbba668e7b7a14